podpac/core/coordinates/rotated_coordinates.py

```diff
@@ -1,11 +1,210 @@
+        if val[0] <= 0 or val[1] <= 0:
+            raise ValueError("Invalid shape %s, shape must be positive" % (val,))
+        return val
+
+    @tl.validate('step')
+    def _validate_step(self, d):
+        val = d['value']
+        if val[0] == 0 or val[1] == 0:
+            raise ValueError("Invalid step %s, step cannot be 0" % val)
+        return val
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Alternate Constructors
+    # ------------------------------------------------------------------------------------------------------------------
+
+    @classmethod
+    def from_geotransform(cls, geotransform, shape, dims=None,
+                               ctypes=None, units=None, segment_lengths=None, coord_ref_sys=None):
+        affine = rasterio.Affine.from_gdal(*geotransform)
+        ulc = affine.c, affine.f
+        deg = affine.rotation_angle
+        scale = ~affine.rotation(deg) * ~affine.translation(*ulc) * affine
+        step = np.array([scale.a, scale.e])
+        return cls(shape, np.deg2rad(deg), ulc, step, dims=dims,
+                   ctypes=ctypes, units=units, segment_lengths=segment_lengths, coord_ref_sys=coord_ref_sys)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Serialization
+    # ------------------------------------------------------------------------------------------------------------------
+
+    @property
+    def definition(self):
+        d = OrderedDict()
+        d['shape'] = self.shape
+        d['theta'] = self.theta
+        d['ulc'] = self.ulc
+        d['step'] = self.step
+        d['dims'] = self.dims
+        d.update(self.properties)
+        return d
+
+    @classmethod
+    def from_definition(cls, d):
+        if 'shape' not in d:
+            raise ValueError('RotatedCoordinates definition requires "shape" property')
+        if 'theta' not in d:
+            raise ValueError('RotatedCoordinates definition requires "theta" property')
+        if 'ulc' not in d:
+            raise ValueError('RotatedCoordinates definition requires "ulc" property')
+        if 'step' not in d and 'lrc' not in d:
+            raise ValueError('RotatedCoordinates definition requires "step" or "lrc" property')
+        if 'dims' not in d:
+            raise ValueError('RotatedCoordinates definition requires "dims" property')
+
+        shape = d['shape']
+        theta = d['theta']
+        ulc = d['ulc']
+        kwargs = {k:v for k,v in d.items() if k not in ['shape', 'theta', 'ulc']}
+        return RotatedCoordinates(shape, theta, ulc, **kwargs)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # standard methods
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def __repr__(self):
```

Comment:
this is a different __repr__ than DependentCoordinates which it wraps. Just making sure this is on purpose

Reply:
Yes, intentional, but I should go back and check that this all looks reasonable when it is in alone and when it is in a Coordinates object.